app/controllers/network_router_controller.rb

```diff
@@ -451,14 +391,16 @@ def form_external_gateway(params)
         subnet = find_record_with_rbac(CloudSubnet, params[:cloud_subnet_id])
         options[:external_gateway_info][:external_fixed_ips] = [{:subnet_id => subnet.ems_ref}]
       end
-      options[:external_gateway_info][:enable_snat] = switch_to_bool(params[:enable_snat])
+      if params.fetch_path(:extra_attributes, :external_gateway_info, :enable_snat)
+        options[:external_gateway_info][:enable_snat] = params[:extra_attributes][:external_gateway_info][:enable_snat]
+      end
     end
     options
   end
 
   def form_params(params)
-    options = %i(name ems_id admin_state_up cloud_group_id cloud_subnet_id
-      cloud_network_id).each_with_object({}) do |param, opt|
+    options = %i(name admin_state_up ems_id cloud_group_id cloud_subnet_id
```

Comment:
`ems_id`? Is this still being used? Because if so, you can't be using `ext_management_system.id` for the same purpose :)

Reply:
Not using `ems_id. I replaced it, thanks.